sympy/solvers/solvers.py

```diff
@@ -1705,7 +1705,15 @@ def _expand(p):
         try:
             soln = _tsolve(f_num, symbol, **flags)
             if soln is not None:
-                result = soln
+                for solnArgs in soln:
+                    if solnArgs.has(LambertW) and solnArgs.has(Lambda):
+                        return soln
+                        break
+                    else:
+                        result = soln
+            if result is False and soln == []:
+                return soln
```

Comment:
I don't know what this is doing but I don't think it should be here. Why isn't this just being handled in `_tsolve`?

Reply:
I think it could be removed. This was the case for lambert on lambert case. (But that should be dealt with another way.)